### PR TITLE
feat: use api-sidecar-handler to offload sshkey handling and add more types

### LINKED ISSUE #1584: SSH private key not validated on updateProject
**Describe the bug**
Using the `updateProject` mutation you can set the private key field to _any_ string. This will cause operations using that key to fail.

**To Reproduce**
1. Use the `updateProject` mutation, providing an invalid private key.
2. Mutation succeeds.
3. Inspect the project, and see that it has an invalid private key.

**Expected behavior**
I expected the SSH key to be validated in some way. It doesn't even have to be fully parsed if that is expensive, just check that the first line is the correct format.

**Screenshots**
n/a

**Additional context**
n/a

### LINKED ISSUE #2189: Support ECDSA SSH keys and FIDO/U2F key types in Lagoon
> The YubiKey 5 Series supports the following algorithms on the PIV smart card application.
>
>    RSA 1024
>    RSA 2048
>    ECC P-256
>    ECC P-384

https://support.yubico.com/support/solutions/articles/15000014219-yubikey-5-series-technical-manual

I'd like to use my Yubikey to interact with Lagoon via SSH, but only RSA and Ed25519 keys are supported by Lagoon. For various reasons RSA 2048-bit keys aren't great so it would be nice to add support for at least `ecdsa-sha2-nistp256`, and possibly `ecdsa-sha2-nistp384`.

### PR DESCRIPTION
**Pull request by shreddedbacon**
# General Checklist

- [ ] Affected Issues have been mentioned in the Closing issues section
- [ ] Documentation has been written/updated
- [ ] PR title is ready for inclusion in changelog

# Database Migrations

- [ ] If your PR contains a database migation, it **MUST** be the latest in date order alphabetically

The node package `sshpk` doesn't support all the various key types out there. The aim of this PR is to introduce a sidecar to required services that runs a simple http server that can replace the functionality of the `sshpk` package.

Each `api` and `webhooks2tasks` pod will get this sidecar so that they aren't reliant on a single service for this functionality, and as there is no state it doesn't need to be a single service and fits nicely as a sidecar.

It introduces new APIs to interact with user ssh keys that doesn't require the user to define the type independently. The input is just `publicKey` which can be the full `ssh-ed25519 A....z` format public key (including comment). The database table for the type is also converted from `enum` to `string` to support any new types, and since the older APIs had enum enforcement, that still applies for anyone that uses the older APIs.

The old APIs remain for now, but have been flagged as deprecated. They will be removed in a future release to give time for tooling and UI to be updated to use the newer APIs.

### Add public key
```
addUserSSHPublicKey(input: {
  name: "my-users-publickey-example"
  publicKey: "ssh-rsa AAAAB3NzaC1yc2EA<SNIP>LOiWw=="
  user: {
    email: "my-user@example.com"
  }
}) {
  id
}
```
### Update public key
```
updateUserSSHPublicKey(input: {
  id: 1
  patch: {
    publicKey: "ssh-rsa AAAAB3NzaC1yc2EA<SNIP>LOiWw=="
  }
}) {
  id
}
```
### Delete public key
```
deleteUserSSHPublicKey(input: {
    id: 1
})
```

# Closing issues

closes #2189 
closes #1584 
addresses #2384 by deprecating `deleteSshKey` which takes the `name` input

### CODE REVIEW

File: services/api/database/migrations/20240508000000_sshkey.js
Comment (tobybellwood) [inline services/api/database/migrations/20240508000000_sshkey.js:1]: I merged (out of order) `services/api/database/migrations/20240512000000_sshkeyused.js` so this migration will need renaming after that :see_no_evil:

Comment (shreddedbacon) [reply services/api/database/migrations/20240508000000_sshkey.js:1]: all good, will fix

File: services/api/src/routes/keys.ts [hunk 1]
```diff
@@ -82,7 +81,7 @@ const keysRoute = async (
     );
   }
 
-  res.send(result);
```
Comment (tobybellwood) [inline services/api/src/routes/keys.ts:85]: This errors here when tried to access with a key thats not in Lagoon (and this trips up anything using the ssh agent)
```
[2024-06-30 23:15:51] [user_auth]: Legacy authentication granted for 'ssh' from 'unknown': {"user":{"id":"ssh","role":"admin","aud":"api.dev","iat":1719789351,"permissions":{}},"headers":{"user-agent":"Wget/1.21.2","content-type":"application/json","content-length":"69","host":"api:3000","ipAddress":"::ffff:172.22.0.23"},"payload":null}
[2024-06-30 23:15:51] [debug]: Accessing keys with fingerprint: SHA256:Kc6CB00BxZHGzQC4UwXSOFsdM3c27HPe4bmpp8hiF0Q
[2024-06-30 23:15:51] [debug]: Unknown fingerprint: SHA256:Kc6CB00BxZHGzQC4UwXSOFsdM3c27HPe4bmpp8hiF0Q
[2024-06-30 23:15:51] [error]: uncaughtException: Cannot read properties of undefined (reading 'key')
TypeError: Cannot read properties of undefined (reading 'key')
    at /app/services/api/dist/routes/keys.js:100:23
    at Generator.next (<anonymous>)
    at fulfilled (/app/services/api/dist/routes/keys.js:28:58)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
[2024-06-30 23:15:51] [error]: uncaughtException: Cannot read properties of undefined (reading 'key')
TypeError: Cannot read properties of undefined (reading 'key')
    at /app/services/api/dist/routes/keys.js:100:23
    at Generator.next (<anonymous>)
    at fulfilled (/app/services/api/dist/routes/keys.js:28:58)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
[winston] Unknown logger level: error
```

Comment (shreddedbacon) [reply services/api/src/routes/keys.ts:85]: ugh merges

Comment (tobybellwood) [reply services/api/src/routes/keys.ts:85]: theres also an issue with existing keys being shown as unknown (but still proceeding)
```
[2024-06-30 23:28:55] [debug]: Accessing keys with fingerprint: SHA256:inQGcrMz0Bp0fTovkhOQgH70z8sMU8jjZbrHSw2MPN4
[2024-06-30 23:28:55] [debug]: Unknown fingerprint: SHA256:inQGcrMz0Bp0fTovkhOQgH70z8sMU8jjZbrHSw2MPN4
```

Comment (shreddedbacon) [reply services/api/src/routes/keys.ts:85]: You may need to pull the changes, the `res.send(result);` was removed already

Comment (shreddedbacon) [reply services/api/src/routes/keys.ts:85]: Oh wait :see_no_evil:

File: services/api/database/migrations/20240730000000_sshkey_types.js
Comment (tobybellwood) [inline services/api/database/migrations/20240730000000_sshkey_types.js:1]: this migration is named in the future :scream: - we can probably settle on 0630?

Comment (shreddedbacon) [reply services/api/database/migrations/20240730000000_sshkey_types.js:1]: ah yeh my bad, it was meant to be 0630

**Review by tobybellwood — APPROVED** [review]
TEsted thoroughly and nits worked out. Good to go

**Pull request merged**